src/mono/wasm/runtime/memory.ts

```diff
@@ -408,3 +408,11 @@ export function isSharedArrayBuffer(buffer: any): buffer is SharedArrayBuffer {
     // See also https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toStringTag
     return sharedArrayBufferDefined && buffer[Symbol.toStringTag] === "SharedArrayBuffer";
 }
+
+export function forceThreadMemoryViewRefresh() {
+    // this condition should be eliminated by rollup on non-threading builds
+    if (!MonoWasmThreads) return false;
+    const wasmMemory = Module.getMemory();
+    wasmMemory.grow(0);
+    runtimeHelpers.updateMemoryViews();
```

Comment:
Without updating the effect does not get propagated? What is the role of update exactly?

Reply:
Ok, I see, it's how they do it, the re-allocate the heaps: https://github.com/emscripten-core/emscripten/blob/a2f3b925c967b9a8091b5b52fa8cbb649cfe248e/src/preamble.js#L146

Reply:
`wasmMemory.grow` always invalidates existing views